src/map.cpp

```diff
@@ -419,7 +419,7 @@ void Map::getSpectators(SpectatorVec& spectators, const Position& centerPos, boo
 			if (it != playersSpectatorCache.end()) {
 				if (!spectators.empty()) {
 					const SpectatorVec& cachedSpectators = it->second;
-					spectators.insert(spectators.end(), cachedSpectators.begin(), cachedSpectators.end());
+					spectators.addSpectators(cachedSpectators);
```

Comment:
I'm not sure, but I think you could even save an extra line, if you do it like this:
```cpp
spectators.addSpectators(static_cast<const SpectatorVec&>(it->second));
```
it is right?

Reply:
I am not using my suggestions!
I am using exactly the changes that you post, and I have a problem with the function: `canSee`, apparently the spectators  they don't stay alive on the memory, causing a problem when obtaining for example the position:

Here I give a small explanation, at first I thought it was because of other changes, but I definitely checked 100% that the culprit is these changes:

https://github.com/otland/forgottenserver/pull/3272#pullrequestreview-582063937

Reply:
They don't, `SpectatorVec` just holds (non-owning) pointers, your creatures are being deleted elsewhere.

Reply:
Just change to: `spectators.addSpectators(it->second);` since the same function converts the value into a read-only reference automatically, I have tried it and if it works, the conversion actually happens.